app/services/conditions_not_met.rb

```diff
@@ -1,11 +1,14 @@
 class ConditionsNotMet
   include ActiveModel::Validations
 
-  def initialize(application_choice:)
+  def initialize(actor:, application_choice:)
+    @auth = ProviderAuthorisation.new(actor: actor)
     @application_choice = application_choice
   end
 
   def save
+    @auth.assert_can_make_offer!(application_choice: @application_choice, course_option_id: @application_choice.offered_option.id)
```

Comment:
What do you think about renaming this (and its sister `can_make_offer?`, and the class `OfferAuthorisation`) `assert_can_make_decisions` etc? cc @tijmenb

Reply:
Makes sense, but also fine to do in a separate PR.

Reply:
I think it's the right thing to do, adding to this PR. `OfferAuthorisation` is only called from `ProviderAuthorisation` and the rest is just a global search/replace for `can_make_offer`.